### ISSUE TITLE
Table with replica in status `erasing_data` incorrectly marked as not ready.

### ISSUE THREAD

**Issue by VeXocide** [rethinkdb/rethinkdb#3478]
When reducing the replicas of a table the table is marked as not ready, such as in

```
[
    {
        "db": "test",
        "id": "9b42f3a5-f7a9-4862-8de3-0da737129b15",
        "name": "test",
        "shards": [
            {
                "primary_replica": "newton_hk3",
                "replicas": [
                    {
                        "server": "newton_hk3",
                        "state": "ready"
                    },
                    {
                        "server": "newton_9m0",
                        "state": "erasing_data"
                    }
                ]
            }
        ],
        "status": {
            "all_replicas_ready": false,
            "ready_for_outdated_reads": true,
            "ready_for_reads": true,
            "ready_for_writes": true
        }
    }
]
```

though all replicas you need are ready, an old one is simply being removed.

**Comment by danielmewes** [comment]
Ok, let's do this:
* hide `erasing_data` replicas from the `replicas` lists
* make `all_replicas_ready` ignore `erasing_data` replicas as well

I'm also generally on board with adding the `role` field back to the replicas in `table_status`. What would its values be? "nothing", "primary replica", "secondary replica"?
It seems that the `role` field was removed as part of https://github.com/rethinkdb/rethinkdb/issues/3064#issue-42936490 ? In that case I would like to wait on @coffeemug 's opinion regarding adding it back.

**Comment by danielmewes** [comment]
Assigning @Tryneus

**Comment by coffeemug** [comment]
I originally asked to remove the `role` field because the wording was very confusing. For example, what does it mean to have `{role: 'nothing'}`? (I know what it means, but to a user, it's very confusing when this shows up)

I think with a few changes in wording, this could be a helpful field. I propose the following:

```
{ goal: 'become_replica' }
{ goal: 'become_primary_replica' }
{ goal: 'recuse_responsibilities' }
```

I would drop the `goal` field entirely when the goal is satisfied -- this would be a transient field that shows up when the replica is in a process of transition.

Note, I don't think this is particularly important to add, but I'm not opposed to adding it.

**Comment by danielmewes** [comment]
Thanks for chiming in @coffeemug. Seems like we should give the details a bit more though. Let's skip the `goal`/`role` field for 1.16.
(but still do the `erasing_data` change)

**Comment by Tryneus** [comment]
The fix for ignoring replicas in the `erasing_data` state has been approved and merged to `next` in commit 2122c51.  Review was 2466.  Will be in release 1.16.
